localstack-core/localstack/aws/handlers/validation.py

```diff
@@ -18,16 +19,43 @@
 from localstack.aws.chain import Handler, HandlerChain
 from localstack.constants import INTERNAL_RESOURCE_PATH
 from localstack.http import Response
+from localstack.utils.analytics.usage import UsageSetCounter
 
 LOG = logging.getLogger(__name__)
 
 
+class UsageCollectorFactory:
+    _collector_registry: dict[str, Any] = {}
+    """Registry for the different paths."""
+
+    NAMESPACE_PREFIX = "internal"
+    """Namespace prefix to track usage of public endpoints (_localstack/ and _aws/)."""
+
+    @classmethod
+    def get_collector(cls, path: str):
+        namespace = f"{cls.NAMESPACE_PREFIX}/{path}"
+        if namespace not in cls._collector_registry:
+            cls._collector_registry[namespace] = UsageSetCounter(namespace)
+        return cls._collector_registry[namespace]
+
+
 class OpenAPIValidator(Handler):
     open_apis: list["OpenAPI"]
 
     def __init__(self) -> None:
         self._load_specs()
 
+    def _record_usage(self, context: RequestContext) -> None:
+        if config.DISABLE_EVENTS:
+            return
+
+        request_path = context.request.path
```

Comment:
I'm a bit worried about the path part, as for some routes, we have dynamic path variables.

And API Gateway is technically part of the internal routes now, even though it serves a pretty different purpose (providing a path variant of the host endpoint) than our other internal routes which provide LocalStack only features. 
Also, API Gateway can have many different parts, which makes it even trickier, we will create a lot, lot of Counters .